Automattic/jetpack
# Search: Restore filter query string functionality

Fixes a regression introduced by #21275.

#### Changes proposed in this Pull Request:
* Fixes a bug where clicking on a filter link would spawn the search interface without setting the appropriate filters in the query string.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Before switching to this code branch, try the following on a site with a Jetpack Search subscription with instant search active.
	1. Add a Jetpack Search widget with filters to any of your sidebars outside of the Search Overlay widget sidebar.
	2. Ensure that the filters appear on your sidebar of choice, like so: <img width="380" alt="Screen Shot 2021-10-29 at 3 57 07 PM" src="https://user-images.githubusercontent.com/4044428/139506095-f3caeb00-9c4b-4b93-afc9-4a67fab03922.png">
	3. Click on one of your filters.
	4. Notice that only `s=` query string has been applied.
* Now, try steps 1~3 with changes from this PR. Ensure that a filter query string as been applied in addition to the `s=` (e.g. `category=classic&s=`).
